test/testdata/parser/error_recovery/unop_binop.rb

```diff
@@ -0,0 +1,68 @@
+# typed: false
+
+class A
+  def g
+    t = T.let(true, T::Boolean)
+    f = T.let(false, T::Boolean)
+    puts (0 .. def) # error: unexpected token ")"
```

Comment:
What's the intuition for why we're testing `def` here? This doesn't look like something anyone would ever write—I'm wondering what the intuition is for including it.